### PR TITLE
Admin: Redirect Jetpack Admin Pages to Specific Sections

### PR DESCRIPTION
**Pull request by lezama**
When linking to a specific page with a hash (#), (ie. https://wp.org/wp-admin/admin.php?page=jetpack#/newsletter) the hash gets stripped by the login redirect. As a result, these pages can’t be linked from documentation or support resources. This change ensures there's a way to link users to specific pages.

## Proposed changes:

•Query Parameter Check: Verify if the page query parameter is set and starts with jetpack/.
•Section Extraction: Extract the specific section from the page parameter.
•Safe Redirect: Perform a safe redirect to the Jetpack admin page, anchoring the user to the specified section.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1719856268371719-slack-C052XEUUBL4

## Testing instructions:

* Go to https://yoursite.org/wp-admin/admin.php?page=jetpack/newsletter
* Check that you get redirected to https://yoursite.org/wp-admin/admin.php?page=jetpack#/newsletter
* Make sure other pages work

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 1]
```diff
@@ -75,6 +75,16 @@ public function __construct() {
 	 * @param Jetpack $jetpack object.
 	 */
 	public function on_jetpack_loaded( $jetpack ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Comment (oskosk) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php]: Maybe move to https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php#L52

Comment (zinigor) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php]: Do we need the ignore twice in this function? We can disable/enable it for this code block, but please specify why as a short comment:
` phpcs:ignore WordPress.Security.NonceVerification.Recommended -- a GET request that doesn't change anything on the server side`

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 2]
```diff
@@ -75,6 +75,16 @@ public function __construct() {
 	 * @param Jetpack $jetpack object.
 	 */
 	public function on_jetpack_loaded( $jetpack ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended		
+		if ( isset( $_GET['page'] ) ) {
+			$page = sanitize_text_field( wp_unslash( $_GET['page'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			if ( strpos( $page, 'jetpack/' ) === 0 ) {
+				$section = substr( $page, 8 );
```
Comment (zinigor) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php]: Why is offset set to `8` here? Or is it just a POC for one page? :)

Comment (lezama) [reply projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php]: to retrieve what's after `jetpack/`

Comment (zinigor) [reply projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php]: Ah, OK, my bad, I forgot that it's the offset, not the length. This line could use a comment for folks like me :)

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php (hunk 1 above)
Comment (lezama) [reply projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php]: > Do we need the ignore twice in this function?

phpcs complains if we don't add it twice.

Comment (lezama) [reply projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php]: good call

**Review by enejb — DISMISSED** [review]
The code looks good 

I tested different cases and they all worked as expected for me.

**Review by enejb — APPROVED** [review]
:+1:

**Pull request merged**